tests/unit/test_workflow_serialization.py

```diff
@@ -123,16 +124,21 @@ def test_save_workflow(self):
     @ordered
     def test_load_workflow(self):
         '''Test loading a workflow from yaml:'''
-        from gquant.dataframe_flow.workflow import load_workflow
+        from gquant.dataframe_flow import TaskGraph
         workflow_file = os.path.join(self._test_dir, 'test_save_workflow.yaml')
 
         with open(workflow_file, 'w') as wf:
             wf.write(WORKFLOW_YAML)
 
-        task_list = load_workflow(workflow_file)
+        task_list = TaskGraph.load_taskgraph(workflow_file)
         all_tasks_exist = True
         for itask in self._task_list:
-            if itask not in task_list:
+            match = False
+            for t in task_list:
+                if itask['id'] == t['id']:
```

Comment:
This is only comparing the `id` field. What about the other fields in the tasks. The logic previously:
```
if itask not in task_list:
```
Automatically finds the matching dict and compares all fields if such a dict exists.
```
In [25]: d1 = {'id': 'sameid', 'conf': {'path': 'diffpath1'}}
In [26]: d2 = {'id': 'sameid', 'conf': {'path': 'diffpath2'}}
In [27]: d2['id'] == d1['id']
Out[27]: True
In [28]: d2 in [d1]
Out[28]: False
In [29]: d3 = deepcopy(d2)
In [30]: d2 in [d1, d3]
Out[30]: True
```
This is a bug. Should be fixed.

Reply:
now check all the fields

Reply:
https://github.com/yidong72/gQuant/blob/fcade400d5c70c6b29a607407d3c69b83ac8cd5a/tests/unit/test_workflow_serialization.py#L138-L148

Does this need to be so verbose. You can just rely on dict comparison as part of the dict feature in the python language. 
This seems overly complex. Just do the `not in` comparison:
```
        task_list = TaskGraph.load_taskgraph(workflow_file)
        match = True
        match = False if len(task_list) != len(self._task_list) else True
        if match:
            for itask in self._task_list:
                if itask not in task_list:
                    match = False
                    break

```

Reply:
done.